phoenix-core/src/test/java/org/apache/phoenix/index/IndexToolSingleRowVerifyTest.java

```diff
@@ -0,0 +1,634 @@
+        when(rebuildScanner.setIndexKeyToMutationMap(Matchers.<Map>any())).thenCallRealMethod();
+        rebuildScanner.setIndexTableTTL(HConstants.FOREVER);
+        indexMaintainer = pIndexTable.getIndexMaintainer(pDataTable, pconn);
+        rebuildScanner.setIndexMaintainer(indexMaintainer);
+    }
+
+    private void initializeGlobalMockitoSetup() throws IOException {
+        //setup
+        when(rebuildScanner.getIndexRowKey(put)).thenCallRealMethod();
+        when(rebuildScanner.prepareIndexMutations(put, delete)).thenCallRealMethod();
+        when(rebuildScanner.verifySingleIndexRow(Matchers.<Result>any(),
+                Matchers.<IndexToolVerificationResult.PhaseResult>any())).thenCallRealMethod();
+        doNothing().when(rebuildScanner)
+                .logToIndexToolOutputTable(Matchers.<byte[]>any(),Matchers.<byte[]>any(),
+                Mockito.anyLong(),Mockito.anyLong(), Mockito.anyString(),
+                        Matchers.<byte[]>any(), Matchers.<byte[]>any());
+        doNothing().when(rebuildScanner)
+                .logToIndexToolOutputTable(Matchers.<byte[]>any(),Matchers.<byte[]>any(),
+                Mockito.anyLong(),Mockito.anyLong(), Mockito.anyString());
+
+        //populate the local map to use to create actual mutations
+        indexKeyToMutationMapLocal = Maps.newTreeMap(Bytes.BYTES_COMPARATOR);
+        rebuildScanner.setIndexKeyToMutationMap(indexKeyToMutationMapLocal);
+        rebuildScanner.prepareIndexMutations(put, delete);
+
+        //populate map to use in test code
+        Map<byte[], List<Mutation>> indexKeyToMutationMap = Maps.newTreeMap((Bytes.BYTES_COMPARATOR));
+        rebuildScanner.setIndexKeyToMutationMap(indexKeyToMutationMap);
+        rebuildScanner.prepareIndexMutations(put, delete);
+    }
+
+    private byte[] getValidRowKey() {
+        return indexKeyToMutationMapLocal.entrySet().iterator().next().getKey();
+    }
+
+    @Test
+    public void testVerifySingleIndexRow_validIndexRowCount_nonZero() throws IOException {
+        IndexToolVerificationResult.PhaseResult expectedPR = getValidPhaseResult();
+        for (Map.Entry<byte[], List<Mutation>>
+                entry : indexKeyToMutationMapLocal.entrySet()) {
+            initializeLocalMockitoSetup(entry, TestType.VALID);
+            //test code
+            rebuildScanner.verifySingleIndexRow(indexRow, actualPR);
+            //assert
+            actualPR.equals(expectedPR);
+        }
+    }
+
+    @Test
+    public void testVerifySingleIndexRow_validIndexRowCount_moreActual() throws IOException {
+        IndexToolVerificationResult.PhaseResult expectedPR = getValidPhaseResult();
+        for (Map.Entry<byte[], List<Mutation>>
+                entry : indexKeyToMutationMapLocal.entrySet()) {
+            initializeLocalMockitoSetup(entry, TestType.VALID_MORE_MUTATIONS);
+            //test code
+            rebuildScanner.verifySingleIndexRow(indexRow, actualPR);
+            //assert
```

Comment:
nit: don't have to tell me you assert just before a line called assert. :-) Ditto for the other places this got pasted.